tests/lib/rules/no-unsafe-optional-chaining.js

```diff
@@ -0,0 +1,375 @@
+            "(obj?.foo ?? baz) ** bar;",
+            "void obj?.foo;",
+            "typeof obj?.foo;",
+            "!obj?.foo",
+            "~obj?.foo",
+            "+(obj?.foo ?? bar)",
+            "-(obj?.foo ?? bar)",
+            "bar |= obj?.foo;",
+            "bar &= obj?.foo;",
+            "bar ^= obj?.foo;",
+            "bar <<= obj?.foo;",
+            "bar >>= obj?.foo;",
+            "bar >>>= obj?.foo;",
+            "bar ||= obj?.foo",
+            "bar &&= obj?.foo",
+            "bar += (obj?.foo ?? baz);",
+            "bar -= (obj?.foo ?? baz)",
+            "bar *= (obj?.foo ?? baz)",
+            "bar /= (obj?.foo ?? baz)",
+            "bar %= (obj?.foo ?? baz);",
+            "bar **= (obj?.foo ?? baz)",
+
+            `async function foo() {
+              (await obj?.foo || baz) + bar;
+              (await obj?.foo ?? baz) + bar;
+              (await obj?.foo ?? baz) - bar;
+              (await obj?.foo ?? baz) * bar;
+              (await obj?.foo ?? baz) / bar;
+              (await obj?.foo ?? baz) % bar;
+              "(await obj?.foo ?? baz) ** bar;",
+              "void await obj?.foo;",
+              "typeof await obj?.foo;",
+              "!await obj?.foo",
+              "~await obj?.foo",
+              "+(await obj?.foo ?? bar)",
+              "-(await obj?.foo ?? bar)",
+              bar |= await obj?.foo;
+              bar &= await obj?.foo;
+              bar ^= await obj?.foo;
+              bar <<= await obj?.foo;
+              bar >>= await obj?.foo;
+              bar >>>= await obj?.foo
+              bar += ((await obj?.foo) ?? baz);
+              bar -= ((await obj?.foo) ?? baz);
+              bar /= ((await obj?.foo) ?? baz);
+              bar %= ((await obj?.foo) ?? baz);
+              bar **= ((await obj?.foo) ?? baz);
+            }`
+        ].map(code => ({
+            code,
+            options: [{
+                disallowArithmeticOperators: true
+            }]
+        })),
+        {
+            code: "obj?.foo - bar;",
+            options: [{}]
+        },
+        {
+            code: "obj?.foo - bar;",
+            options: [{
+                disallowArithmeticOperators: false
+            }]
+        }
+    ],
+
+    invalid: [
+        ...[
+            "(obj?.foo)();",
+            "(obj.foo ?? bar?.baz)();",
+            "(obj.foo || bar?.baz)();",
+            "(obj?.foo && bar)();",
+            "(bar && obj?.foo)();",
+            "(obj?.foo?.())();",
+            "(obj?.foo).bar",
+            "(obj?.foo)[1];",
+            "(obj?.foo)`template`",
+            "new (obj?.foo)();",
+            "new (obj?.foo?.())()",
+            "new (obj?.foo?.() || obj?.bar)()",
+
+            `async function foo() {
+              (await obj?.foo)();
+            }`,
+            `async function foo() {
+              (await obj.foo ?? bar?.baz)();
+            }`,
+            `async function foo() {
+              (await obj?.foo || bar?.baz)();
+            }`,
```

Comment:
I think these two examples are not testing `await`, if that was the intention?

`await obj.foo ?? bar?.baz` is like `(await obj.foo) ?? bar?.baz`